### ISSUE TITLE
Avoid need to write dumb getters/setters - a decorator, direct access,..? 

### ISSUE THREAD

**Issue by breznak** [numenta/nupic#2674]
So..I got tired of writing simple getters/setters in obvious cases, like:
```
self.x = 1
def setX(self, x):
  self.x=x
def getX(self):
  return self.x
```
Of course it has to be done for more complex cases with some computation, checks, etc. 

Read here: https://stackoverflow.com/questions/6618002/python-property-versus-getters-and-setters 
It seems the "pythonic way" would be to differentiate variables by underscore in name, w/o underscore means "public" and is a part of the API. The just use direct access for the variables.

That would be nice, but I see the problem in `c++/swig` seamless implementation - is there a way to fix this for that case? 

An option would be to create a decorator that creates a get/set function for that member. 

What do you think? 
@scottpurdy @rhyolight @oxtopus

**Comment by scottpurdy** [comment]
I don't think we should use a decorator that adds methods. I am fine with public attributes / member variables where they make sense, both in Python and C++.
